src/maker/api.rs

```diff
@@ -92,6 +91,48 @@ pub struct ConnectionState {
     pub pending_funding_txes: Vec<Transaction>,
 }
 
+pub struct ThreadPool {
+    pub threads: Mutex<Vec<JoinHandle<()>>>,
+}
+
+impl Default for ThreadPool {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Drop for ThreadPool {
+    fn drop(&mut self) {
+        if let Err(e) = self.join_all_threads() {
+            log::error!("Error joining threads in via drop: {:?}", e);
+        }
+    }
+}
+
+impl ThreadPool {
+    pub fn new() -> Self {
+        Self {
+            threads: Mutex::new(Vec::new()),
+        }
+    }
+
+    pub fn add_thread(&self, handle: JoinHandle<()>) {
+        let mut threads = self.threads.lock().unwrap();
+        threads.push(handle);
+    }
+    #[inline]
+    fn join_all_threads(&self) -> Result<(), MakerError> {
+        let mut threads = self
+            .threads
+            .lock()
+            .map_err(|_| MakerError::General("Failed to lock threads"))?;
+        while let Some(thread) = threads.pop() {
+            thread.join().unwrap();
```

Comment:
Instead of unwrap, how about just `log::error` the error. The error later is being suppressed anyway in the drop implementation, so we can as well just log it here. 

Currently, this loop will return at the first instance of the error and will not try to join the rest of the threads. So lets just log the error here if any, or else just say "Thread X terminated successfully".

Reply:
Yep, makes sense.